plutus-contract/src/Plutus/Trace/Scheduler.hs

```diff
@@ -291,35 +290,35 @@ runThreads e = do
                     & enqueue (suspendThread Normal initialThread)
 
 -- | Run the threads that are scheduled in a 'SchedulerState' to completion.
-loop :: forall effs systemEvent.
+loop :: forall a effs systemEvent.
     ( Eq systemEvent
     , Member (LogMsg SchedulerLog) effs
     )
-    => SchedulerState effs systemEvent
-    -> Eff effs ()
+    => SchedulerState effs systemEvent a
+    -> Eff effs (Maybe a)
```

Comment:
How do you know that result here is the one from the initialThread and not from the wallet threads?

Reply:
You only get a result from an `exit` call. There's only one place where `exit` is called and that's in `handleEmulatorTrace` which calls it with the result of the incoming `action`.

Reply:
Then should we mention this somewhere? Comment on `runThreads`?